libs/libGamedata/gameData/SelectionMapInputData.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+#include <boost/filesystem.hpp>
+#include <string>
+#include <vector>
+
+struct ImageResource
+{
+    boost::filesystem::path filePath;
+    unsigned index;
+    ImageResource(boost::filesystem::path path = boost::filesystem::path(), unsigned index = 0)
+        : filePath(std::move(path)), index(index){};
+};
+
+struct MissionSelectionInfo
+{
+    ///  The maskAreaColor is the color for the mission used in the missionMapMask
+    unsigned maskAreaColor = 0;
+    /// The position on which the conquered image and the marker image, if mission is selected,
+    /// are displayed. The offset is always counted from the origin of the map image.
+    Position ankerPos;
+};
+
+struct SelectionMapInputData
+{
+    SelectionMapInputData() = default;
+
+    /// Background image for the selection map
+    ImageResource background;
+    /// The map image itself.
+    ImageResource map;
+    /// The image must be the same size as map image. This image describes the mission areas of map image.
+    /// This is used for selection of the missions when the user clicks. Each mission area must have an
+    /// unique color. The color is specified in the missionSelectionInfos in the maskAreaColor parameter.
+    ImageResource missionMapMask;
+    /// The marker image shown when a mission is selected.
+    ImageResource marker;
+    /// The image shown when a mission is already finished.
+    ImageResource conquered;
+    /// offset of the map image and missionMapMask image relative to the background image.
+    /// Can be (0,0) if no offset exists.
+    Position mapOffsetInBackground;
```

Comment:
Maybe combine this doc into an example:
```suggestion
    Position mapOffsetInBackground = {0, 0};
```

This makes it clear that "(0,0)" is accepted and removes the redundant explanation that an offset of 0,0 is "no offset"